rocketchat/app.yaml

```diff
@@ -0,0 +1,549 @@
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+          - name: MONGODB_REPLICA_SET_MODE
+            value: "primary"
+          - name: MONGODB_REPLICA_SET_NAME
+            value: "rs0"
+          - name: MONGODB_ADVERTISED_HOSTNAME
+            value: "$(MONGODB_POD_NAME).rocketchat-mongodb-headless.rocketchat.svc.cluster.local"
+          - name: MONGODB_USERNAME
+            value: "rocketchat"
+          - name: MONGODB_DATABASE
+            value: "rocketchat"
+          - name: MONGODB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: rocketchat-mongodb
+                key: mongodb-password
+          - name: MONGODB_ROOT_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: rocketchat-mongodb
+                key: mongodb-root-password
+          - name: MONGODB_REPLICA_SET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: rocketchat-mongodb
+                key: mongodb-replica-set-key
+          - name: MONGODB_ENABLE_IPV6
+            value: "no"
+          - name: MONGODB_ENABLE_DIRECTORY_PER_DB
+            value: "no"
+          livenessProbe:
+            exec:
+              command:
+                - pgrep
+                - mongod
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            timeoutSeconds: 5
+            successThreshold: 1
+            failureThreshold: 6
+          readinessProbe:
+            exec:
+              command:
+                - mongo
+                - --eval
+                - "db.adminCommand('ping')"
+            initialDelaySeconds: 5
+            periodSeconds: 10
+            timeoutSeconds: 5
+            successThreshold: 1
+            failureThreshold: 6
+          volumeMounts:
+            - name: datadir
+              mountPath: /bitnami/mongodb
+              subPath: 
+          resources:
+            {}
+      volumes:
+  volumeClaimTemplates:
+    - metadata:
+        name: datadir
+        annotations:
+      spec:
+        accessModes:
+          - "ReadWriteOnce"
+        resources:
+          requests:
+            storage: "8Gi"
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: rocketchat-mongodb-secondary
+  namespace: rocketchat
+  labels:
+    app: mongodb
+    chart: mongodb-7.14.8
+    release: rocketchat
+spec:
+  selector:
+    matchLabels:
+      app: mongodb
+      release: rocketchat
+      component: secondary
+  podManagementPolicy: "Parallel"
+  serviceName: rocketchat-mongodb-headless
+  replicas: 0
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: mongodb
+        chart: mongodb-7.14.8
+        release: rocketchat
+        component: secondary
+    spec:
+      serviceAccountName: rocketchat-mongodb
+      securityContext:
+        fsGroup: 1001      
+      initContainers:
+      containers:
+        - name: mongodb-secondary
```

Comment:
Not so certain if we should non-transparently create a DB for Rocket.chat, though I think we could improve the marketplace experience by making applications with interdependencies.

Reply:
so what do you suggest ?

Reply:
Can you check other apps manifest file and check the dependency section ?

Reply:
yes, added

Reply:
you still have mongoDB as a statefulset why is that ?

Reply:
meaning? isn't DB's should be in statefulsets ?

Reply:
So the whole point of adding a dependency in the manifest is that it will automatically be installed from there and you don't have to have a separate installation for it in your app.yaml file, Kindly check an app that has mongodb as dependency , maybe wordpress